### PR TITLE
Fix #2069 -- proper locale support for section links

### LINKED ISSUE #2069: post.section_link ignores languages
In [11]: site.timeline[30].section_link('pl')
Out[11]: '/nieskategoryzowane/'

The correct output should be '/pl/nieskategoryzowane/'. I’m using sections from metadata, but I think this is a problem with the path-based implementation, too.  I’ll attempt to fix it by myself.

### PR DESCRIPTION
**Pull request by Kwpolska**
This is #2069.

cc @Aeyoun, @ralsina.

Signed-off-by: Chris Warrick <kwpolska@gmail.com>

### CODE REVIEW

File: nikola/post.py [hunk 1]
```diff
@@ -745,11 +745,16 @@ def section_color(self, lang=None):
 
     def section_link(self, lang=None):
```
Comment (felixfontein) [inline nikola/post.py]: Why isn't this simply using the `index_section_path` path handler of the `indexes` plugin?

Comment (Kwpolska) [inline nikola/post.py]: That’s a good question. I just fixed the existing code (by @Aeyoun). I looked a bit further, and it seems that the Post  class does not have access to the Nikola site object, and therefore cannot talk to `site.link()` nor to the plugin manager.

Comment (felixfontein) [inline nikola/post.py]: Maybe we should deprecate this function and encourate people to use `site.link()` instead. A `Post` object should know as little as possible about what other plugins are producing (well, except the `pages` and the `posts` plugins).

Comment (ralsina) [inline nikola/post.py]: +1 to just using link. We don't need to have the post access the Nikola object either, tests just need to be resolved after they are written in the document, when copying to output.

**Pull request merged**